### ISSUE TITLE
build: Investigate aarch64 pointer authentication

### ISSUE THREAD

**Issue by dongcarl** [bitcoin/bitcoin#19075]
This would entail adding `-mbranch-protection=standard` to our hardening flags. We should also check if this feature requires kernel/glibc support that our ABI doesn't include.

More info: https://fedoraproject.org/wiki/Changes/Aarch64_PointerAuthentication

**Comment by willcl-ark** [comment]
Has this been completed as part of https://github.com/bitcoin/bitcoin/pull/32071/changes/00ba3ba30341a9073049125334f176d6c05d1b54 ?

ISTM that that change resulted in our current `aarch64/arm64` cmake configuration:

https://github.com/bitcoin/bitcoin/blob/6d625af2831b7c36d59f245a65df3f70f2110f69/CMakeLists.txt#L532-L538

...where we use `bti` on `Darwin` and `standard` on all other `aarch64/arm64` systems.

**Comment by fanquake** [comment]
> Has this been completed as part of https://github.com/bitcoin/bitcoin/commit/00ba3ba30341a9073049125334f176d6c05d1b54 ?

No, but I do have a branch that does complete this, which I'll open this week.

**Comment by fanquake** [comment]
Sorry, thought this was (#30677 which I do have a branch to finish), my PR to finish fixing this issue was #24123, which I closed, in favour of #25573, given that it requires a glibc bump to `2.33`+. If do we end up bumping our glibc in the interim, then this would still be solved.
